lib/src/barback/dartdevc/dartdevc_environment.dart

```diff
@@ -253,24 +255,26 @@ AssetId _entrypointDartId(AssetId id) {
 class _AssetCache {
   /// [Asset]s are indexed first by package and then path, this allows us to
```

Comment:
have you measured the performance difference against
```
for(var assetId in _assets.keys) {
  if(assetId.package == packageName) _assets.remove(packageName);
}
```

It makes the rest of the class a lost simpler if we avoid the nested map...